pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+operating system that will be used.
+
+
+Valid values: `ubuntu1804`
+
+
+Example: `os_image = ubuntu1804`
+
+#### Constants
+
+##### The e1-standard-2 constant
+
+The `e1-standard-2` value used in the preamble represents the hardware of
+the Virtual Machine used for each job in the current pipeline and is a global
+variable.
+
+
+##### The ubuntu1804 constant
+
+The `ubuntu1804` value used in the preamble represents the Operating System of
+the Virtual Machine used for each job in the current pipeline and is a
+global variable.
+
+
+
+### Preamble example
+
+    version: "v1.0"
+    name: "The name of the Semaphore 2.0 project"
+    agent:
+      machine:
+        type: e1-standard-2
+        os_image: ubuntu1804
+    blocks:
+
+
+## Blocks
+
+The `blocks` property is an array of `task` items that hold the elements of
+a `.semaphore/semaphore.yml` file.
+
+The values of the `blocks` property are the tasks of a pipeline as
+defined by the `task` property.
+
+If you accidentally name two or more `blocks` items with the same name,
+you will get an error message similar to the following:
+
+> semaphore.yml ERROR:
+
+> Error: "There are at least two blocks with same name: Build Go project"
+
+
+### name
+
+The `name` property is a Unicode string that assigns a name to the Semaphore
+2.0 project and is optional.
+
+
+### task
+
+All the items of a `blocks` list of of the `task` type.
+
+You will learn about the properties of a `task` item in a while.
+
+
+### Example of Blocks
+
+    blocks:
+     - name: Inspect Linux environment
+       task:
+          jobs:
+            - name: Execute hw.go
+              commands:
+                - echo $SEMAPHORE_PIPELINE_ID
+                - echo $HOME
+                - echo $SEMAPHORE_GIT_DIR
+                - pwd
+                - ls -al
+                - cat hw.go
+                - echo $VAR1
+                - echo $PI
+          prologue:
+              commands:
+                - checkout
+          env_vars:
+               - name: VAR1
+                 value: Environment Variable 1
+               - name: PI
+                 value: "3.14159"
+
+
+## Task
+
+The `task` keyword that comes after the `blocks` property divides a YAML
+configuration file into major and distinct sections. Each `task` item can
+have multiple `jobs` items, an optional `prologue` section, an optional
+`epilogue` section as well as an optional `env_vars` block for defining
+environment variables and an optional `secrets` block for using existing
+`secrets` environment variables.
+
+#### jobs
+
+The `jobs` items of the `blocks` list allow you to define the commands that
+you want to execute.
+
+
+#### secrets
+
+The `secrets` property is used for using existing `secrets` environment
+variables.
+
+
+#### prologue
+
+As the `prologue` block is executed before a job, the result of the job that
+follows that `prologue` block has no effect on the execution of
+the `prologue` block.
```

Comment:
- prologue commands are part of the job.
- they are prepended to all jobs defined in a task
- they are part of all jobs

I don't understand this:

```
`prologue` block has no effect on the execution of
 the `prologue` block.
```